### DISCUSSION TITLE
Running install script for anvil-zksync... ERROR: Unsupported architecture detected!

### DISCUSSION THREAD

**Discussion by Djougabench** [Cyfrin/moccasin-full-course-cu#18] [General]
HI ! Anybody  running that error, when trying to run the  install.sh script ?

**Comment by EngrPips** [comment]
Hello @Djougabench, Have you resolved this already? I ran into the same issue

**Reply by Djougabench** [reply]
hi @EngrPips nope not yet  :-(

**Comment by s3bc40** [comment]
@EngrPips @Djougabench I think there is a recent issue opened on the `anvil-zksync` project:
https://github.com/matter-labs/anvil-zksync/issues/519

The user seems to have a Mac. On my end on Linux I had no trouble.

Maybe the solution will be available when the issue is solved

**Reply by EngrPips** [reply]
Thanks so much, @s3bc40, for the concern. I actually figured out a solution, which is to download the binary and move it to the appropriate folder, and then request my device to allow the binary to run.

**Reply by s3bc40** [reply]
Good to hear 🙏

**Reply by Djougabench** [reply]
i work with ubuntu linux

**Reply by EngrPips** [reply]
@s3bc40 @Djougabench  is working on Linux and they are running into issue maybe you have suggestion or guide for them.

**Reply by s3bc40** [reply]
@Djougabench It might come from your ubuntu architecture, maybe by default the installation script run on `x86_64`...

Mine is `x86_64` which means I am running on a `64bits`. I have installed the release manually on my part since their last release was not working on my side. It is the [v0.2.1](https://github.com/matter-labs/anvil-zksync/releases/tag/v0.2.1), and I followed their tuto on their README. 

There is also a [discussion](https://github.com/Cyfrin/moccasin-full-course-cu/discussions/30) that I have opened explaining how to install `anvil-zksync` manually after downloading their binary. But I'll try to adapt with your problem:

1. Try to run on your terminal:
```sh
uname -m
```

You can check the response from this unix stack exchange thread: https://unix.stackexchange.com/questions/12453/how-to-determine-linux-kernel-architecture

2. Download the right binary depending on your architecture
![image](https://github.com/user-attachments/assets/0802033e-dcd1-4a3c-a95a-22beeaa46035)

- `x86_64` -> the first one
- others -> the second

3. Run this command where you downloaded the binary with the right name 
```sh
# Just replace the tar.gz by the one you download
tar xz -f anvil-zksync-v0.2.1-x86_64-unknown-linux-gnu.tar.gz -C /usr/local/bin/         
chmod +x /usr/local/bin/anvil-zksync
```

It should be good then. But tell us if it does not work, I'll try to find a solution.

Just in case, did you do their [prerequisites](https://github.com/matter-labs/anvil-zksync#-prerequisites)?

**Reply by s3bc40** [reply]
There is also this issue closed two days ago but the issue was on Mac: https://github.com/matter-labs/anvil-zksync/issues/519
